pkg/services/apiserver/builder/admission.go

```diff
@@ -0,0 +1,43 @@
+package builder
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+)
+
+const PluginName = "GrafanaAdmission"
+
+type builderAdmission struct {
+	validators map[schema.GroupVersion]APIGroupValidation
+}
+
+var _ admission.ValidationInterface = (*builderAdmission)(nil)
+
+func (b *builderAdmission) Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	if v, ok := b.validators[a.GetResource().GroupVersion()]; ok {
+		return v.Validate(ctx, a, o)
+	}
+	return nil
+}
+
+func (b *builderAdmission) Handles(operation admission.Operation) bool {
+	return true
+}
+
+func NewAdmissionFromBuilders(builders []APIGroupBuilder) *builderAdmission {
+	validators := make(map[schema.GroupVersion]APIGroupValidation)
+	for _, builder := range builders {
```

Comment:
this was the missing piece for me, how we would register every validator :smile: nice work!